src/main/java/seedu/duke/commands/ModuleShow.java

```diff
@@ -0,0 +1,28 @@
+package seedu.duke.commands;
+
+import seedu.duke.parser.Parser;
+import seedu.duke.storage.Storage;
+import seedu.duke.ui.Ui;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class ModuleShow extends Command{
+    private ArrayList<String> moduleList = new ArrayList<String>(Arrays.asList("show", "CS2113T", "CG2028", "GES1011", "CGF1002"));
+    private String action = "";
+
+    public ModuleShow (String input){
+        this.action = Parser.parseModuleCode(input);
+        try {
+            if (!moduleList.contains(action)){
+                throw new Exception();
+            }
+        } catch (Exception e){
+            Ui.printInvalidModuleCodeMessage();
+        }
+    }
```

Comment:
don't really get what action is for, maybe separate functionality into classes? 

e.g `moduleList` , `moduleAdd` --> we parse in `Parser` and execute specific commands rather than within a custom class, maybe use `moduleListCommand` in `commands` instead